tests/parsing/parchmint-parser.test.js

```diff
@@ -0,0 +1,74 @@
+const ParchmintParser = require('../../parsing/parchmint-parser.js');
+const Layer = require('../../model/layer.js');
+
+const validParchmintLayers = '"layers": [\n' +
+        '    {\n' +
+        '        "id": "unique-flow-layer-id-string",\n' +
+        '        "name": "flow-layer"\n' +
+        '    },\n' +
+        '    {\n' +
+        '        "id": "unique-control-layer-id-string",\n' +
+        '        "name": "control-layer"\n' +
+        '    }\n' +
+        ']';
+
+const invalidParchmintLayers = '"layers": [\n' +
+        '    {\n' +
+        '        "id": "unique-flow-layer-id-string",\n' +
+        '        "name": ""\n' +
+        '    },\n' +
+        '    {\n' +
+        '        "id": "",\n' +
+        '        "name": "control-layer"\n' +
+        '    }\n' +
+        ']';
+
+function parseJSONObj(str) {
+    return JSON.parse('{' + str + '}');
+}
+
+describe('layers', () => {
+    test('initialize with initParchKey', () => {
+        let pp = new ParchmintParser();
+        let l = new Layer();
+
+        pp.initParchKey(l, parseJSONObj(validParchmintLayers).layers[0]);
```

Comment:
You should validate the layer length before requesting the 0th object

Reply:
I validated with ```expect``` and ```toBe``` calls. Is that ok?

Reply:
I would say it depends on what type of error you get from the expect if there isn't an element available. Just comment out the parsing and pass in an empty variable and if you can see that the error is because there is no element vs. the matching wasn't correct then its fine, otherwise you should explicitly check for it.

Reply:
It throws a TypeError if the variable is undefined, and the test fails. If the lengths don't match then the test fails normally.